thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js

```diff
@@ -172,54 +194,84 @@ export default Ember.Component.extend({
    * @param {String} attribute - name of attribute to check against config's input's labelMapping
    */
   isInConfig(attribute) {
-    return this.get('config').some(filterBlock => filterBlock.inputs.some(input => attribute === input.labelMapping));
+    return this.get('filterBlocks').some(filterBlock => filterBlock.inputs.some(input => attribute === input.labelMapping));
   },
 
-  actions: {
+  /**
+   * Computes the filtered urns when there are changes to entities, filterCache, and eventType
+   */
+  filteredUrns: Ember.computed(
+    'entities',
+    'filterCache',
+    'eventType',
+    function () {
+      const { entities, filterCache, eventType } =
+        this.getProperties('entities', 'filterCache', 'eventType');
 
-    /**
-     * Expands/collapses a filter block
-     * @method filterByEvent
-     * @param {Object} clickedBlock - selected filter block object
-     */
-    selectEventType(clickedBlock) {
-      const { entities, onSelect } = this.getProperties('entities', 'onSelect');
-      const cachedEventType = this.urnsCache[clickedBlock.eventType];
-      let filterBlocks = this.get('config');
+      if (!eventType) { return []; }
 
-      // Hide all other blocks when one is clicked
-      filterBlocks.forEach(block => {
-        Ember.set(block, 'isHidden', true);
+      const filters = filterCache[eventType] || {};
+
+      const filteredUrns = Object.keys(entities).filter(urn => {
+        const e = entities[urn];
+        return e.type == 'event' && e.eventType == eventType && this.applyFilters(e, filters);
       });
 
-      // Note: toggleProperty will not be able to find 'filterBlocks', as it is not an observed property
-      // Show clickedBlock
-      Ember.set(clickedBlock, 'isHidden', !clickedBlock.isHidden);
+      return filteredUrns;
+    }
+  ),
 
-      /*
-      * If results were already previously filtered for this filter block (i.e. "anomaly", "holiday"),
-      * call onSelect on the cached urns
-      */
-      if (!Ember.isEmpty(cachedEventType)) {
-        onSelect(cachedEventType);
-      }
-      // If this is the first time results are computed, cache them
-      else {
-        const urns = Object.keys(entities).filter(urn => entities[urn].type == 'event'
-                                                  && entities[urn].eventType == clickedBlock.eventType);
-        this.urnsCache[clickedBlock.eventType] = urns;
-        onSelect(urns);
+  /**
+   * Determines whether to apply filters
+   * @param {Object} entity
+   * @param {Object} filters - subfilters selected (i.e. {countryCode: 'DE'})
+   */
+  applyFilters(e, filters) {
+    if (_.isEmpty(filters)) {
+      return true;
+    }
+
+    return Object.keys(filters).every(dimName => {
```

Comment:
nit: let's use parenthesis around the argument names when the array function spans over multiple line.